mm2src/coins/eth.rs

```diff
@@ -856,96 +864,101 @@ impl SwapOps for EthCoin {
         amount: &BigDecimal,
         min_block_number: u64,
         _uuid: &[u8],
-    ) -> Box<dyn Future<Item = (), Error = String> + Send> {
+    ) -> Box<dyn Future<Item = (), Error = MmError<ValidateSwapTxError>> + Send> {
         let selfi = self.clone();
         let tx = match fee_tx {
             TransactionEnum::SignedEthTx(t) => t.clone(),
             _ => panic!(),
         };
-        let sender_addr = try_fus!(addr_from_raw_pubkey(expected_sender));
-        let fee_addr = try_fus!(addr_from_raw_pubkey(fee_addr));
+        let sender_addr = try_mm_err_fus!(addr_from_raw_pubkey(expected_sender));
+        let fee_addr = try_mm_err_fus!(addr_from_raw_pubkey(fee_addr));
         let amount = amount.clone();
 
         let fut = async move {
-            let expected_value = try_s!(wei_from_big_decimal(&amount, selfi.decimals));
-            let tx_from_rpc = try_s!(
-                selfi
-                    .web3
-                    .eth()
-                    .transaction(TransactionId::Hash(tx.hash))
-                    .compat()
-                    .await
-            );
+            let expected_value = wei_from_big_decimal(&amount, selfi.decimals)?;
+            let tx_from_rpc = selfi
+                .web3
+                .eth()
+                .transaction(TransactionId::Hash(tx.hash))
+                .compat()
+                .await?;
             let tx_from_rpc = match tx_from_rpc {
                 Some(t) => t,
-                None => return ERR!("Didn't find provided tx {:?} on ETH node", tx),
+                None => {
+                    return MmError::err(ValidateSwapTxError::InternalError(format!(
```

Comment:
First of all, I think here we need to return a `ValidateSwapTxError::TxIsMissing("Didn't find provided tx {:?} on ETH node")` instead of `InternalError`. It's not our error, the other party could send false data.
Secondly, you can use `or_mm_err` instead of pattern matching.

Reply:
done